File: ui/composite/TrackToolBar.py
__author__ = 'toramisu'
from  ui import *
from module.Events import *


class TrackToolBar(QWidget):
    def __init__(self, parent):
        super(TrackToolBar, self).__init__(parent)
        self.resize(250, 25)
        self.newTrackButton = QPushButton(self)
        self.newTrackButton.setText('new Track')
        self.newTrackButton.mousePressEvent = self.onPressNewTrack

    def onPressNewTrack(self, e):
        Event.dis(ActionEvent.NEW_TRACK, 'new layer')
        pass

File: view/timeline/TrackToolBar.cpp
//
// Created by toramisu on 2015/6/16.
//

#include <model/App.h>
#include "TrackToolBar.h"
#include "model/TrackModel.h"
static void someFunction() {
//    qDebug() << "pressed";
    printf("click me");
}
void TrackToolBar::onClkNewTrackBtn() {
    printf("pressed");

}

TrackToolBar::TrackToolBar(QWidget *parent):QWidget(parent)
{
    resize(250, 25);
    newTrackButton = new QPushButton("new Track",this);
//    QObject::connect(newTrackButton, &QPushButton::clicked, someFunction);
//    newTrackButton->mouseMoveEvent
//    TrackModel *t = TrackModel::_();
//    t->newTrack();
    App()._().trackModel->newTrack("curry","D:/projects/SeqTruanCpp/test/test10/");
    connect(newTrackButton, &QPushButton::clicked, []{
        printf("click me");
        Evt_dis("type");
    });
//    this.pressed.connect(someFunction);
//    this->onClkNewTrackBtn();
}

